### PR TITLE
STSMACOM-620: Add additional prop in SearchAndSort

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
https://issues.folio.org/browse/STSMACOM-620

*Problem*

We need to have the ability to change paneSubTitle when props change and change the text in the button in SearchFiled. All that needs to be done in the scope of the Browse call numbers feature.

*Solution*

Simple add additional props in SearchAndSort components.

### CODE REVIEW

File: lib/SearchAndSort/SearchAndSort.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ class SearchAndSort extends React.Component {
     columnWidths: PropTypes.object,
     createRecordPath: PropTypes.string,
     customPaneSub: PropTypes.node,
+    customPaneSubText: PropTypes.string,
```
Comment (zburke) [inline lib/SearchAndSort/SearchAndSort.js:93]: Are you 100% confident this should be `string`? To accept a `<FormattedMessage>` I would have expected `node`, or perhaps `oneOfType([PropTypes.string, PropTypes.element])`.

File: lib/SearchAndSort/SearchAndSort.js [hunk 2]
```diff
@@ -211,6 +212,7 @@ class SearchAndSort extends React.Component {
     resultsVirtualize: PropTypes.bool,
     searchableIndexes: PropTypes.arrayOf(PropTypes.object),
     searchableIndexesPlaceholder: PropTypes.string,
+    searchFiledButtonLabel: PropTypes.string,
```
Comment (zburke) [inline lib/SearchAndSort/SearchAndSort.js:215]: Typo, I think: `s/Filed/Field/`.

File: lib/SearchAndSort/SearchAndSort.js [hunk 3]
```diff
@@ -1058,7 +1060,7 @@ class SearchAndSort extends React.Component {
           disabled={!searchTerm}
           data-test-search-and-sort-submit
         >
-          <FormattedMessage id="stripes-smart-components.search" />
+          <FormattedMessage id={this.props.searchFiledButtonLabel || 'stripes-smart-components.search'} />
```
Comment (zburke) [inline lib/SearchAndSort/SearchAndSort.js:1063]: All other `this.props` values are destructured above beginning on line 1008/1010. Please follow the established convention. And as above, fix the typo: `s/Filed/Field/`.

File: lib/SearchAndSort/SearchAndSort.js [hunk 4]
```diff
@@ -1184,6 +1187,9 @@ class SearchAndSort extends React.Component {
             <FormattedMessage id={messageKey} values={{ count: source.totalCount() }} />
           )}
           {!isSourceLoaded && (
+            customPaneSubText?
```
Comment (zburke) [inline lib/SearchAndSort/SearchAndSort.js:1190]: Run lint before pushing up your branch to avoid these kinds of errors.

**Review by zburke — CHANGES_REQUESTED** [review]

File: lib/SearchAndSort/SearchAndSort.js (hunk 2 above)
Comment (UladzislauKutarkin) [reply lib/SearchAndSort/SearchAndSort.js:215]: What do you mean` s/Filed/Field/`?

Comment (UladzislauKutarkin) [reply lib/SearchAndSort/SearchAndSort.js:215]: Thanks a lot, my little mistake

File: lib/SearchAndSort/SearchAndSort.js (hunk 3 above)
Comment (UladzislauKutarkin) [reply lib/SearchAndSort/SearchAndSort.js:1063]: Done.

File: lib/SearchAndSort/SearchAndSort.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply lib/SearchAndSort/SearchAndSort.js]: I add oneOfType.

Comment (UladzislauKutarkin) [reply lib/SearchAndSort/SearchAndSort.js]: You are right, It's better to use node to accept a `<FormattedMessage>`. So, I change a logic a little bit and now we can use node.

File: lib/SearchAndSort/SearchAndSort.js (hunk 4 above)
Comment (UladzislauKutarkin) [reply lib/SearchAndSort/SearchAndSort.js]: Done.

**Review by JohnC-80 — APPROVED** [review]
Looks good. Thanks for addressing Zak's changes.

**Review by zburke — APPROVED** [review]

**Pull request merged**